ui/src/app/oauth.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+
+@Injectable({
+  providedIn: "root",
+})
+export class OauthService {
+  endpoint = "http://localhost:8080/api/session";
```

Comment:
maybe not a point for now, but later, won't this need to be changed?

Reply:
yeah we'll need to introduce actual environmental build variables but we can cross that bridge in a tiny bit. We can write the url from the build script so I'll wait to introduce this abstraction until we're a little further down :)